gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -321,8 +321,10 @@ protected void submitFlowToDagManagerHelper(DagActionStore.DagAction dagAction,
       launchSubmissionMetricProxy.markFailure();
       return;
     } catch (SpecNotFoundException e) {
-      log.warn("Spec not found for flowId {} due to exception {}", flowId, e.getMessage());
-      launchSubmissionMetricProxy.markFailure();
+      log.info("Spec not found for flowId {} due to deletion by active dagManager host due to exception {}",
+          flowId, e.getMessage());
+      // TODO: mark this failure if there are other valid cases of this exception
+      // launchSubmissionMetricProxy.markFailure();
```

Comment:
1. let's figure out whether there are other valid cases!

not sure if I'm misunderstanding... but a TODO is far more reasonable when the impl change would be considerable effort, so we plan to come back, vs. when the task is to figure out how our system behaves.  usually, the latter is worth resolving ASAP.

give that, was it incorrect previously to `markFailure()` and we no longer want to do?

2. on the message: `"Spec not found for flowId {} due to deletion by active dagManager host due to exception {}",`

is that certain it was due to deletion by active DM or is that speculative?

Reply:
It's speculative, I think there could be a case where a SpecNotFoundException occurs in a case _not_ due to deletion by active DM if something goes wrong at a different point in the system (mistaken deletion NOT from the deleteFlowSpecIfAdhoc code path). It's hard to definitively say all of these errors are due to an active DM deleting it but I would expect them all to be.